dm/dm/master/scheduler/scheduler.go

```diff
@@ -671,6 +682,7 @@ func (s *Scheduler) transferWorkerAndSource(lworker, lsource, rworker, rsource s
 
 // TransferSource unbinds the `source` and binds it to a free or same-source-relay `worker`.
 // If fails halfway, the old worker should try recover.
+// if input worker is empty means try transfer this source to a free worker.
```

Comment:
what's the scenario we didn't specify a worker? and how do we tell the transfer is in processing or transfered to the old worker?